### PR TITLE
gen: update then template to match new action

### PR DESCRIPTION
**Pull request by noamcattan**
example yaml file:
```yaml
name: Atlas
on:
  push:
    branches:
      - master
    paths:
      - .github/workflows/ci-atlas.yaml
      - 'migrations/*'
  pull_request:
    paths:
      - 'migrations/*'
# Permissions to write comments on the pull request.
permissions:
  contents: read
  pull-requests: write
jobs:
  atlas:
    services:
      # Spin up a mysql:8 container to be used as the dev-database for analysis.
      mysql:
        image: mysql:8
        env:
          MYSQL_DATABASE: dev
          MYSQL_ROOT_PASSWORD: pass
        ports:
          - 3306:3306
        options: >-
          --health-cmd "mysqladmin ping -ppass"
          --health-interval 10s
          --health-start-period 10s
          --health-timeout 5s
          --health-retries 10
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
        with:
          fetch-depth: 0
      - uses: ariga/setup-atlas@v0
        with:
          cloud-token: ${{ secrets.ATLAS_CLOUD_TOKEN }}
      - uses: ariga/atlas-action/migrate/lint@master
        with:
          dir: 'file://migrations'
          dir-name: 'name'
          dev-url: 'mysql://root:pass@localhost:3306/dev'
      - uses: ariga/atlas-action/migrate/push@master
        if: github.ref == 'refs/heads/master'
        with:
            dir: 'file://migrations'
            dir-name: 'name'
            dev-url: 'mysql://root:pass@localhost:3306/dev'
            cloud-token: ${{ secrets.ATLAS_CLOUD_TOKEN }}

```

### CODE REVIEW

File: gen/atlas.tmpl [hunk 1]
```diff
@@ -11,33 +14,29 @@ permissions:
...
-          cloud-token: {{`${{ secrets.`}}{{ .SecretName }}{{` }}`}}
-  sync:
-    needs: lint
-    {{- template "services" . }}
-    if: github.ref == 'refs/heads/{{ .DefaultBranch }}'
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v3
-      - uses: ariga/atlas-sync-action@v0
+      - uses: ariga/atlas-action/migrate/push@master
```
Comment (dorav) [inline gen/atlas.tmpl:34]: Should be using `v1` by default

```suggestion
      - uses: ariga/atlas-action/migrate/push@v1
```

File: gen/atlas.tmpl [hunk 2]
```diff
@@ -11,33 +14,29 @@ permissions:
...
-          name: '{{ . }}'
-{{- end }}
-          {{ template "UseServices" (args .Driver .Driver) }}
-          cloud-token: {{`${{ secrets.`}}{{ .SecretName }}{{` }}`}}
+            dir: 'file://{{ .Path }}'
+            {{- with .DirName }}
+            dir-name: '{{ . }}'
+            {{- end }}
+            {{ template "UseServices" (args .Driver "localhost") }}
+            cloud-token: {{`${{ secrets.`}}{{ .SecretName }}{{` }}`}}
```
Comment (dorav) [inline gen/atlas.tmpl:42]: this parameter moved to the setup-atlas action

File: gen/testdata/postgres.yml [hunk 3]
```diff
@@ -32,36 +35,18 @@ jobs:
...
+        if: github.ref == 'refs/heads/master'
         with:
-          dir: 'migrations'
-          name: 'name'
-          dev-url: 'postgres://postgres:pass@postgres:5432/dev?search_path=public&sslmode=disable'
-          cloud-token: ${{ secrets.ATLAS_CLOUD_TOKEN }}
\ No newline at end of file
+            dir: 'file://migrations'
+            dir-name: 'name'
+            dev-url: 'postgres://postgres:pass@localhost:5432/dev?search_path=public&sslmode=disable'
```
Comment (dorav) [inline gen/testdata/postgres.yml:51]: why did the host for the driver change to localhost?
Do you remember why it was different for lint and sync?

Comment (dorav) [reply gen/testdata/postgres.yml:51]: There was added complexity added to support passing two parameters to the template, we can remove that extra complexity if we can always use `localhost`

Comment (noamcattan) [reply gen/testdata/postgres.yml:51]: The reason it was not localhost is that we ran the previous action in docker. Now we run on "bare metal" so localhost is good.

File: gen/atlas.tmpl (hunk 1 above)
Comment (noamcattan) [reply gen/atlas.tmpl:34]: didn't we agree on master, and that it will fetch the latest?

Comment (ronenlu) [reply gen/atlas.tmpl:34]: I think that is better to specify explicit version, but not crucial IMO

File: gen/testdata/postgres.yml (hunk 3 above)
Comment (dorav) [reply gen/testdata/postgres.yml:51]: gotcha, so let's remove all the "args" thing that I added to support passing two parameters?

File: gen/atlas.tmpl (hunk 1 above)
Comment (dorav) [reply gen/atlas.tmpl:34]: as far as I understood from my conversations with Rotem, using v{major} is the way to go for pulling latest action on github actions ecosystem (you will automatically get minor versions, but nothing breaking)

File: gen/atlas.tmpl [hunk 4]
```diff
@@ -3,6 +3,9 @@ on:
   push:
     branches:
       - {{ .DefaultBranch }}
+    paths:
+      - .github/workflows/ci-atlas.yaml
+      - '{{ .Path }}/*'
```
Comment (dorav) [inline gen/atlas.tmpl:8]: shouldn't these paths only be inside the `pull_request` and `master branch` triggers?

**Review by dorav — DISMISSED** [review]

**Review by dorav — APPROVED** [review]

**Pull request merged**